src/hotspot/os/bsd/os_bsd.cpp

```diff
@@ -2328,13 +2328,53 @@ bool os::Bsd::hugetlbfs_sanity_check(bool warn, size_t page_size) {
 
 // Large page support
 
-static size_t _large_page_size = 0;
+static size_t _large_page_size =
+#if defined(__FreeBSD__)
```

Comment:
My comment applies to the whole block here.  (I obviously need to figure out to do that within the Github review system, but since I haven't I am stating it instead)

What assignment does this code make on FreeBSD/i386?  That appears to fall into none of the conditionals here and seems like it would yield a compile error.

Reply:
Nope sorry my bad  I naively thought AMD64 covered it  ...